cloudinit/config/cc_ansible.py

```diff
@@ -0,0 +1,115 @@
+"""ansible enables running on first boot either ansible-pull"""
+
+from copy import deepcopy
+from logging import Logger
+from textwrap import dedent
+from typing import Tuple
+
+from cloudinit.cloud import Cloud
+from cloudinit.config.schema import MetaSchema, get_meta_doc
+from cloudinit.distros import ALL_DISTROS, Distro
+from cloudinit.settings import PER_INSTANCE
+from cloudinit.subp import ProcessExecutionError, subp, which
+
+meta: MetaSchema = {
+    "id": "cc_ansible",
+    "name": "Ansible",
+    "title": "Configure ansible for instance",
+    "description": dedent(
+        """\
+        This module provides ansible integration.``
+
+        Ansible is often used agentless and in parallel
+        across multiple hosts simultaneously. This
+        doesn't fit the model of cloud-init: a single
+        host configuring itself during boot. Instead,
+        this module installs ansible during boot and
+        then uses ``ansible-pull`` to run the playbook
+        repository at the remote URL.
+        """
+    ),
+    "distros": [ALL_DISTROS],
+    "examples": [
+        dedent(
+            """\
+            #cloud-config
+            ansible:
+              install: true
+              pull:
+                url: "https://github.com/holmanb/vmboot.git"
+                playbook-name: ubuntu.yml
+            """
+        ),
+    ],
+    "frequency": PER_INSTANCE,
```

Comment:
Can now add the following to ensure we don't even call this module's handle when no applicable config is provided.
```suggestion
    "activate_by_schema_keys": ["ansible"],
    "frequency": PER_INSTANCE,
```